### ISSUE TITLE
Arguments with Union and Iterable are not added by add_class_arguments

### ISSUE THREAD

**Issue by gatoniel** [omni-us/jsonargparse#53]
Hi,

when using `Union` together with `Iterable` for an argument this argument is not detected by `add_class_arguments`. See the following code example
```python
from typing import Union, Iterable, List
from jsonargparse import ArgumentParser


class OtherClass(object):
    def __init__(self, name: str = "test"):
        self.name = name


class Class1(object):
    def __init__(
        self, other: Union[OtherClass, List[OtherClass], bool] = True
    ):
        self.other = other


class Class2(object):
    def __init__(
        self, other: Union[OtherClass, Iterable[OtherClass], bool] = True
    ):
        self.other = other


if __name__ == "__main__":
    parser1 = ArgumentParser()
    parser2 = ArgumentParser()
    parser1.add_class_arguments(Class1, "Class1")
    parser2.add_class_arguments(Class2, "Class2")

    cfg1 = parser1.parse_args()
    cfg2 = parser2.parse_args()

    print(cfg1)
    print(cfg2)
```
with the following output

> Namespace(Class1=Namespace(other=True))
> Namespace()

I would expect this output

> Namespace(Class1=Namespace(other=True))
> Namespace(Class2=Namespace(other=True))

However, accordingt to the list in https://jsonargparse.readthedocs.io/en/stable/index.html?highlight=class_path#type-hints `Union` together with `Iterable` should be supported.

**Comment by mauvilsa** [comment]
Hi @gatoniel. Thank you for reporting this issue. I had made sure that all of lightning's Trainer init parameters were added to the parser. However, this development was done in python 3.6 and this issue was for python>=3.7 so I didn't notice. This is now fixed and available in jsonargparse v3.10.1.

**Comment by gatoniel** [comment]
Thanks a lot! Works for me!
